### DISCUSSION TITLE
[MultiCompiler] Share asset details between compilers

### DISCUSSION THREAD

**Discussion by phoenisx** [webpack/webpack#15311] [Q&A] [answered]
Hi!

Is it possible to signal/share asset details, like `contenthash` between two different compilers, when using MultiCompiler config?

Is there any hook to do so?
If not, can we achieve it in some other way?

**Comment by alexander-akait** [comment]
Only from stats, in `done` hook

**Reply by phoenisx** [reply]
Is it a good idea to write to file (asset details) from the first compiler's custom plugin, and read from the second compilers custom plugin

**Comment by phoenisx** [comment]
Also, it's really hard to understand whether MultiCompiler works in async or in sync??

<img width="720" alt="image" src="https://user-images.githubusercontent.com/11786283/152346962-b56f7c2b-2ccf-4100-96d8-778fb4172339.png">

In the docs it's mentioned Multiple Configs will be read one at a time, but my logs show weird behaviour.

For e.g. I have 2 configs where first config's entry is `main.js` and second config's entry is `main2.js`, each has a custom plugin, where I am console logging some string for `assetEmitted` Hook.

The desired output should have been:
-> assetEmmited hook output for main.js config
-> assetEmmited hook output for main2.js config
-> build done/complete for main.js
-> build done/complete for main2.js

But the output is :
-> assetEmmited hook output for main2.js config
-> assetEmmited hook output for main.js config (This should've been called first)
-> build done/complete for main.js
-> build done/complete for main2.js

**Comment by phoenisx** [comment, marked as answer]
Closing this discussion as I created a plugin to communicate between 2 compilers.
As of now the plugin just consume data from one compiler to another, but it's safer to wait for the data to be present, which might block one of the compilers running thread.
